pandas/tests/generic/test_generic.py

```diff
@@ -948,3 +948,15 @@ def test_deprecated_get_dtype_counts(self):
         df = DataFrame([1])
         with tm.assert_produces_warning(FutureWarning):
             df.get_dtype_counts()
+
+    def test_boolean_in_replace(self):
+        # GH 27660
+        df = DataFrame({"col": [False, True, 0, 1]})
+
+        result = df.replace({"col": {False: 0, True: 1}})
```

Comment:
If you were to switch the replacement to `{False: 1, True: 0}` then this test will fail as the integers will get incorrectly replaced:
```python
In [2]: df = pd.DataFrame({'col': [False, True, 0, 1]})

In [3]: df.replace({'col': {False: 1, True: 0}})
Out[3]: 
   col
0    1
1    0
2    1
3    0
```
I think the current version of the test is passing because 0/1 are just getting replaced by themselves.

This is a consequence of how Python handles hashing, specifically `0`/`False` have the same hash and evaluate equally with `==` (Python's fallback on hash collision), so they'll be considered the same in `set` operations or when looking up keys in a `dict` (likewise for `1`/`True`).

Reply:
ehh, you are very right, it's not good solution @jschendel